### PR TITLE
Enhance2 parser uname 1x

### PR DESCRIPTION
**Pull request by wushiqinlou**
* parser recognize the line starting with "Linux" as correct input

### CODE REVIEW

**Comment by wushiqinlou** [comment]
@kylape  According to my knowledge, there is no output such as "2.6.32-279.el6.x86_64" from command "uname -a". So I delete test instance "UNAME_ABBREVIATED", and I changed the following code written by you so that code would recognize "2.6.32-279.el6.x86_64" as invalid string.

~~~
if len(uname_parts) < 3:
            ver_rel_match = re.match("[0-9](\.[0-9]+){2}-[0-9]+", uname_parts[0])
            if not ver_rel_match:
                raise UnameError("Uname string appears invalid", uname_line)
            data['kernel'] = uname_parts[0]
~~~

**Comment by wushiqinlou** [comment]
This enhance is applied for two issue:

1, The last line is blank in the output of "uname -a":

~~~
Linux qqhrycsq2 2.6.32-279.el6.x86_64 #1 SMP Wed Jun 13 18:24:36 EDT 2012 x86_64 x86_64 x86_64 GNU/Linux

~~~ 

2, There is another non-blank line in the output of "uname -a":

~~~
COMMAND> uname -a
Linux qqhrycsq2 2.6.32-279.el6.x86_64 #1 SMP Wed Jun 13 18:24:36 EDT 2012 x86_64 x86_64 x86_64 GNU/Linux
~~~

File: insights/parsers/tests/test_uname.py [hunk 1]
```diff
@@ -27,7 +27,15 @@
 
 UNAME_CVE_2016_0728_1 = 'Linux hostname.example.com 3.10.0-229.1.2.rt56.141.2.el7_1.x86_64 #1 SMP Fri Dec 19 12:09:25 EST 2014 x86_64 x86_64 x86_64 GNU/Linux'
 
-UNAME_ABBREVIATED = '3.10.1-327.204.el7.x86_64'
+UNAME_BLANK_LINE = """
```
Comment (bfahr) [inline insights/parsers/tests/test_uname.py]: @PaulWay you added tests for this, can you take a look at this PR please.

File: insights/parsers/uname.py [hunk 2]
```diff
@@ -210,13 +209,13 @@ def parse_content(self, content):
...
+        if not uname_list:
+            raise UnameError("Uname string appears invalid", '')
+        uname_line = uname_list[0]
         uname_parts = uname_line.split(' ')
         if len(uname_parts) < 3:
-            ver_rel_match = re.match("[0-9](\.[0-9]+){2}-[0-9]+", uname_parts[0])
-            if not ver_rel_match:
-                raise UnameError("Uname string appears invalid", uname_line)
-            data['kernel'] = uname_parts[0]
+            raise UnameError("Uname string appears invalid", uname_line)
```
Comment (bfahr) [inline insights/parsers/uname.py]: @wushiqinlou this code is inherited from the original `uname` utility class as is the classmethod `from_kernel` below.  It does allow the creation of uname objects from a shortened uname string.  I'm not convinced that we want to remove this working code unless it is causing problems with the parsing.

Comment (wushiqinlou) [inline insights/parsers/uname.py]: @bfahr, I modified function `from_kernel` and `from_uname_str`, and we can use these two function to create uname objest from a shortened uname string. As no shortened uname string from normal command `uname -a`, we may not delivery shortened uname string to class `Uname` directly.

Comment (bfahr) [inline insights/parsers/uname.py]: @wushiqinlou that will be sufficient for me then, thanks.

**Pull request merged**